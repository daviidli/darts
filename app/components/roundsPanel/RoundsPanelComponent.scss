.rounds {
	padding: 1em;
	padding-top: 2em;
	height: 75vh;
}
