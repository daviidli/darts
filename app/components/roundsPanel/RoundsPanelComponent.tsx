/* eslint-disable react/no-array-index-key */
import React from 'react';
import { Typography } from '@material-ui/core';
import Rounds from '../rounds/Rounds';
import { getPreviousRounds } from '../../utils/dartUtils';
import styles from './RoundsPanelComponent.scss';

export type Props = {
	rounds: string[][][];
	currentPlayer: number;
	currentRound: number;
	maxRounds: number;
};

const RoundsPanelComponent = (props: Props) => {
	const { rounds, currentPlayer, currentRound, maxRounds } = props;

	const roundsShown = 6;
	return (
		<div className={styles.rounds}>
			<Typography variant="h4">
				{`Rounds (${currentRound + 1}/${maxRounds})`}
			</Typography>
			<ol
				start={
					currentRound <= roundsShown
						? 1
						: currentRound - (roundsShown - 1)
				}
			>
				{getPreviousRounds(
					rounds[currentPlayer],
					roundsShown,
					currentRound
				).map((round: string[], i) => (
					<Rounds round={round} key={`round-${i}`} />
				))}
			</ol>
		</div>
	);
};

export default RoundsPanelComponent;
