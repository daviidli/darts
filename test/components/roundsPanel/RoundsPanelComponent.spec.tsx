import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import RoundsPanelComponent, {
	Props
} from '../../../app/components/roundsPanel/RoundsPanelComponent';

Enzyme.configure({ adapter: new Adapter() });

const initialProps: Props = {
	rounds: [[['1']]],
	currentPlayer: 0,
	currentRound: 0,
	maxRounds: 1
};

const setup = (props: Props = initialProps) => {
	const wrapper = shallow(<RoundsPanelComponent {...props} />);
	return { wrapper };
};

describe('RoundsPanel component', () => {
	it('should match snapshot', () => {
		const { wrapper } = setup();
		expect(toJson(wrapper)).toMatchSnapshot();
	});

	it('should match snapshot with 2 rounds', () => {
		const props: Props = {
			rounds: [
				[
					['1', '2', '3'],
					['1-d', '2-t', 'b']
				]
			],
			currentPlayer: 0,
			currentRound: 0,
			maxRounds: 2
		};
		const { wrapper } = setup(props);
		expect(toJson(wrapper)).toMatchSnapshot();
	});

	it('should match snapshot with many rounds', () => {
		const props: Props = {
			rounds: [
				[
					['1', '2', '3'],
					['1-d', '2-t', 'b'],
					['1', '2', '3'],
					['1-d', '2-t', 'b'],
					['1', '2', '3'],
					['1-d', '2-t', 'b'],
					['1', '2', '3'],
					['1-d', '2-t', 'b'],
					['1', '2', '3'],
					['1-d', '2-t', 'b']
				]
			],
			currentPlayer: 0,
			currentRound: 7,
			maxRounds: 9
		};
		const { wrapper } = setup(props);
		expect(toJson(wrapper)).toMatchSnapshot();
	});
});
