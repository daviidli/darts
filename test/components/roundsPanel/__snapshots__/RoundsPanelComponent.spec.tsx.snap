// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoundsPanel component should match snapshot 1`] = `
<div
  className="rounds"
>
  <WithStyles(ForwardRef(Typography))
    variant="h4"
  >
    Rounds (1/1)
  </WithStyles(ForwardRef(Typography))>
  <ol
    start={1}
  />
</div>
`;

exports[`RoundsPanel component should match snapshot with 2 rounds 1`] = `
<div
  className="rounds"
>
  <WithStyles(ForwardRef(Typography))
    variant="h4"
  >
    Rounds (1/2)
  </WithStyles(ForwardRef(Typography))>
  <ol
    start={1}
  />
</div>
`;

exports[`RoundsPanel component should match snapshot with many rounds 1`] = `
<div
  className="rounds"
>
  <WithStyles(ForwardRef(Typography))
    variant="h4"
  >
    Rounds (8/9)
  </WithStyles(ForwardRef(Typography))>
  <ol
    start={2}
  >
    <Rounds
      key="round-0"
      round={
        Array [
          "1-d",
          "2-t",
          "b",
        ]
      }
    />
    <Rounds
      key="round-1"
      round={
        Array [
          "1",
          "2",
          "3",
        ]
      }
    />
    <Rounds
      key="round-2"
      round={
        Array [
          "1-d",
          "2-t",
          "b",
        ]
      }
    />
    <Rounds
      key="round-3"
      round={
        Array [
          "1",
          "2",
          "3",
        ]
      }
    />
    <Rounds
      key="round-4"
      round={
        Array [
          "1-d",
          "2-t",
          "b",
        ]
      }
    />
    <Rounds
      key="round-5"
      round={
        Array [
          "1",
          "2",
          "3",
        ]
      }
    />
  </ol>
</div>
`;
